routers/crudRouter.js

```diff
@@ -36,53 +36,86 @@ exports.getRouter = function (Model) {
   })
 
   router.put('/:id', (req, res) => {
-    MongoClient.connect(MONGU_URI).then(connection => {
-      // TODO: validate
-      connection.db().collection(Model.collectionName).findOneAndUpdate(idMongoQuery(Model.IdField, parseInt(req.params.id)), req.body).then(dbResponse => {
-        if (dbResponse.ok === 1) {
-          res.sendStatus(200)
-        }
+    let validationResult = true
+    if (options.validationMethods.put) {
+      validationResult = options.validationMethods.put(req.params)
+    }
+
+    if (validationResult) {
+      MongoClient.connect(MONGU_URI).then(connection => {
+        connection.db().collection(options.collectionName).findOneAndUpdate(idMongoQuery(options.IdField, parseInt(req.params.id)), req.body).then(dbResponse => {
+          if (dbResponse.ok === 1) {
+            res.sendStatus(204)
+          }
+        })
+      }).catch(err => {
+        MsLogger.error(err)
+        res.sendStatus(500)
       })
-    }).catch(err => {
-      MsLogger.error(err)
-      res.sendStatus(500)
-    })
+    } else {
```

Comment:
Here you put the validation results far from the actual validations. I think it would be more readable if you would do it the other way:
```javascript
if(!validated) {
  res.sendStatus(400)
  return
}

// process if validation succeeded
```